coderd/httpmw/authz.go

```diff
@@ -46,7 +47,13 @@ func AsAuthzSystem(mws ...func(http.Handler) http.Handler) func(http.Handler) ht
 // Requires using a Recorder Authorizer.
 func RecordAuthzChecks(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
-		r = r.WithContext(rbac.WithAuthzCheckRecorder(r.Context()))
+		// Only enabled if the x-authz-checks header is set to a truthy value.
+		// The requester is expected to explicitly ask for this header to be set
+		// in the response.
+		if enabled, _ := strconv.ParseBool(r.Header.Get("x-authz-checks")); enabled {
```

Comment:
Should we make this more explicit, like `x-enable-authz-checks`?

`x-authz-checks` is what gets returned, so it could get confusing.